### ISSUE TITLE
Bulk importing of CSV files

### ISSUE THREAD

**Issue by clibassi** [JuliaData/CSV.jl#712]
Ported from a discussion on Slack - was hoping there might be a way to combine multiple CSV files at once that will ultimately be appended to each other. Current options are multiple CSV.read() statements or: 

`df = CSV.read(firstfile)`
`for file in rest`
    `append!(df, CSV.File(file)`
`end`

**Comment by quinnj** [comment]
Yeah, my thought here is we could have a `CSV.File(::Vector)` method that took a vector of filenames (or `IO` objects, or byte vectors), loop through each to do parsing, and append the results in a single CSV.File w/ `ChainedVector`s for each column, with chains being each file.

**Comment by kmundnic** [comment]
Is having `CSV.File(::Vector)` necessary? I use dot notation for this:

```julia
df = vcat(CSV.read.(files, DataFrame)...)
```

**Comment by rofinn** [comment]
That solution requires a dependency on DataFrames.jl. I think if CSV.jl handled this internally it could be a lot more efficient than splatting the results into a `vcat` call. That operation would be very slow if you had hundreds of csv files. Alternatively, if `Tables.Partitioner` was also a valid table then `CSV.File(::Vector)` could always return a lazy partitioner that you could then be materialized as whatever table type folks want?

**Comment by quinnj** [comment]
DataFrames.jl has a specialized method on `vcat` that makes it pretty efficient. We already have `TableOperations.joinpartitions` that takes a valid `Tables.partitions` input and produces a "table", so you could do `df = Tables.Partitioner(CSV.File, myfiles) |> TableOperations.joinpartitions |> DataFrame`. `joinpartitions` operates lazily.

**Comment by rofinn** [comment]
Okay, I'll open a PR to TableOperations.jl to add that to the README.md

**Comment by rofinn** [comment]
Seems buggy?

```julia
  Got exception outside of a @test
  BoundsError: attempt to access 3-element Vector{AbstractVector{T} where T} at index [4]
  Stacktrace:
    [1] getindex
      @ ./array.jl:801 [inlined]
    [2] getcolumn
      @ ~/.julia/packages/CSV/Zl2ww/src/file.jl:52 [inlined]
    [3] getcolumn
      @ ~/.julia/packages/CSV/Zl2ww/src/file.jl:80 [inlined]
    [4] (::TableOperations.var"#21#24"{Vector{SentinelArrays.ChainedVector}, CSV.File{false}})(i::Int64)
      @ TableOperations ~/.julia/packages/TableOperations/OkUoR/src/TableOperations.jl:298
    [5] foreach(f::TableOperations.var"#21#24"{Vector{SentinelArrays.ChainedVector}, CSV.File{false}}, itr::UnitRange{Int64})
      @ Base ./abstractarray.jl:2141
    [6] joinpartitions(x::Tables.Partitioner{Base.Generator{Vector{CSV.File{false}}, Tables.var"#9#10"{S3DB.Cache.var"#f#31"}}})
      @ TableOperations ~/.julia/packages/TableOperations/OkUoR/src/TableOperations.jl:298
 ```

**Comment by quinnj** [comment]
Ok, PR is up to allow passing a vector of inputs to `CSV.File` or `CSV.read` and have them vertically, lazily concatenated by CSV.jl directly. I outline the reasoning for having this builtin to CSV.jl instead of using generic functionality in the PR. https://github.com/JuliaData/CSV.jl/pull/873

**Comment by quinnj** [comment]
Implemented in #873
